notes/2021/2021-06-03.md

```diff
@@ -0,0 +1,75 @@
+# 2021-06-03 ESLint TSC Meeting Notes
+
+## Transcript
+
+[`2021-06-03-transcript.md`](2021-06-03-transcript.md)
+
+## Attending
+
+* Nicholas C. Zakas (@nzakas) - TSC
+* Milos Djermanovic (@mdjermanovic) - TSC
+* Brandon Mills (@btmills) - TSC
+
+@nzakas moderated, and @btmills took notes and updated issues with resolutions.
+
+## Topics
+
+### [v8.0.0 status](https://github.com/eslint/eslint/projects/8)
+
+#### [Breaking: Ajv to v8.5.0, added ajv-draft-04 (fixes #13888)](https://github.com/eslint/eslint/pull/13911)
+
+* @mdjermanovic feels this would be a breaking change so should be added to the v8.0.0 project. Some of the schema validations are more strict and cannot be turned off, even for end users who have no control over plugin schemas. As a workaround, we could add a `try`/`catch`.
+* @nzakas and @btmills are in favor of catching previously-unreported issues in schemas that could cause them to behave incorrectly.
+* @nzakas wants to be careful not to break the ecosystem. The changes would primarily be 1) adding `type: "object"` or `type: "array"` where they didn't exist previously and 2) removing `additionalItems: false`.
+* @nzakas feels these changes are reasonable and is in favor of adding to v8.0.0. We should check the first prerelease against some popular plugins to confirm the changes are small.
+* @mdjermanovic confirms that end users should not notice the changes, just plugin authors when using `RuleTester`.
```

Comment:
End users should not notice the new validations that will be activated only in `RuleTester`, like missing `type` or ignored `additionalItems: false`, but they might start getting errors in some other cases (e.g., invalid `type` value) unless we add try/catch, which may or may not be a good idea.

Reply:
How's this?

```suggestion
* @mdjermanovic confirms that plugin authors would notice missing `type` or ignored `additionalItems: false` when using `RuleTester`, but end users should only notice problems with their configurations that were previously uncaught. We could add a `try`/`catch` to suppress those, but that may not be a good idea.
```

Reply:
Not exactly, end users can have valid configurations but Ajv will still throw in certain cases where the rule author wrote an invalid schema.

There are 3 major changes in the new Ajv:

1. It's less tolerant.

Ajv 6 can be configured to ignore severe mistakes in schemas, like invalid `type` value (e.g., typo `"type": "arrayy"`). ESLint 7 uses Ajv 6 in a way that catches this kind of errors in `RuleTester`, but not in runtime.

Ajv 7+ throws on an invalid `type` value regardless of the settings. There is no way to disable this check. We could add `try`/`catch` around `ajv.compile()`, but now I think it's a bad idea (https://github.com/eslint/eslint/pull/13911#issuecomment-855390157).

So, if there's a rule with `"type": "arrayy"`, that rule now works with ESLint 7, but will crash with ESLint 8 and there's nothing the end user can do about it until the author fixes the rule and releases a new version.

These mistakes should be rare because `RuleTester` _does_ catch them in ESLint 7.

2. The `strictDefaults` option has been dropped in favor of the new option [`strictSchema`](https://ajv.js.org/strict-mode.html#prohibit-ignored-keywords).

`strictSchema: true` = `strictDefaults: true` + more validations (missing `type` field, nonsensical `additionalItems: false`, etc.).

The new validations helps rule authors to detect schemas that were valid by the specification but probably not working as intended (in particular, they were allowing invalid configurations).

This is the change I was referring to. It will catch _very_ common mistakes, but it will not affect end users when they switch to ESLint v8.0.0 because we're using this option only in `RuleTester`.

Fixing the schemas _may_ affect end users, but only those with invalid configurations. Depending on the plugin's semver policy, the fixes may require major version bump.

3. Ajv 7+ doesn't support JSON Schema draft-04 specification, which is the specification we are using.

This is resolved in https://github.com/eslint/eslint/pull/13911 by replacing `ajv` dependency with [`ajv-draft-04`](https://www.npmjs.com/package/ajv-draft-04), which wraps `ajv` and transforms draft-04 schemas to a format supported by the underlying `ajv`.

Reply:
Something like this (feel free to change the wording):

```suggestion
* @mdjermanovic confirms that end users should not notice new validations that report common mistakes in rule schemas, just plugin authors when using `RuleTester`. End users might start getting errors about more severe mistakes in rule schemas, but it isn't likely that such rules were published because `RuleTester` already reports those errors.
```

Reply:
That’ll do. Thanks!